src/erlcloud_iam.erl

```diff
@@ -981,6 +981,17 @@ data_type("Role") ->
      {"RoleId", role_id, "String"},
      {"RoleName", role_name, "String"},
      {"Path", path, "String"}];
+data_type("RoleGetDetail") ->
```

Comment:
Can you just call this `"RoleDetail"`? Seems like it fits the pattern better.

Reply:
There's already a used by `data_type("RoleDetail")` used by  `data_type("AccountAuthorizationDetails")`. 

I named it as best I could, but wasn't sure. I can change it. Lmk.

Reply:
Didn't realize the conflict. This should work then.

Reply:
Actually, looking at the API docs, it sure looks like there is only [Role](https://docs.aws.amazon.com/IAM/latest/APIReference/API_Role.html) and [RoleDetail](https://docs.aws.amazon.com/IAM/latest/APIReference/API_RoleDetail.html). It looks like both [GetRole](https://docs.aws.amazon.com/IAM/latest/APIReference/API_GetRole.html) and [ListRoles](https://docs.aws.amazon.com/IAM/latest/APIReference/API_ListRoles.html) use `Role`, and [GetAccountAuthorizationDetails](https://docs.aws.amazon.com/IAM/latest/APIReference/API_GetAccountAuthorizationDetails.html) uses `RoleDetails`. It is curious that the `ListRoles` example doesn't have `RoleLastUsed` in the response, but... I bet it's there? Also, it looks like `RoleLastUsed` is part of the `RoleDetail` object as well. Maybe it would be right to just add the `RoleLastUsed` to both the `Role` and `RoleDetail` `data_type` entries. 

Do you have access to an account with IAM to check? I can do it tomorrow if not. I'd prefer to stick with symmetry to the API object structure if the docs are right.

Reply:
Yes, I have an AWS Account and checked the API response. `RoleLastUsed` isn't returned in the [ListRoles](https://docs.aws.amazon.com/IAM/latest/APIReference/API_ListRoles.html#API_ListRoles_Examples) API response, and AWS's docs are correct.

Sorry, I think that I should have noted this originally in the PR. That is why I created a new `data_type`.

Please feel free to double check the AWS API response.

I was also a little confused by this. I first checked `boto3.get_role` and there was a lot more info, but I guess the XML API is a little more specific.

Reply:
OK, I think I have some insight at last:

1. You're right that ListRoles does not return `RoleLastUsed`. I would propose renaming the existing `data_type("Role")` entry to `data_type("RoleList")` or something like that, and making the `get_role` handle the "real" one for `data_type("Role")`. 
2. `"RoleDetail"` used in `get_account_authorization_details()` also returns `RoleLastUsed`, so we should use it there. Can you add it in?

Reply:
I did bullet point 1 here, and the [comment](https://github.com/erlcloud/erlcloud/pull/671#pullrequestreview-546620125) below.

I didn't do bullet point 2 though because `RoleLastUsed` is actually under based on the [API doc](https://docs.aws.amazon.com/IAM/latest/APIReference/API_GetAccountAuthorizationDetails.html): 
```
<InstanceProfileList>
  <member>
    <InstanceProfileName>EC2role</InstanceProfileName>
    <Roles>
      <member>
```

It's under the `InstanceProfileList` which uses `data_type("InstanceProfile")` and that same `data_type` is linked to a couple of other API endpoints for InstanceProfile that don't have a RoleLastUsed field. I left as-is unless it's worth creating a second `data_type("InstanceProfile")` that includes `RoleLastUsed`? Also not sure the `data_type` name?

Reply:
Nah, I guess this is fine. This API is so bizarrely inconsistent.